### PR TITLE
speed up threshold of 64F version using NEON and SSE

### PR DESCRIPTION
**Pull request by vpisarev**
### Former  #6591

use SIMD and speed up #6546 

### What does this PR change?
  * speeds up the threshold function
  * use NEON under aarch64 only

### CODE REVIEW

File: modules/imgproc/src/thresh.cpp [hunk 1]
```diff
@@ -915,6 +915,10 @@ thresh_64f(const Mat& _src, Mat& _dst, double thresh, double maxval, int type)
     size_t src_step = _src.step / sizeof(src[0]);
     size_t dst_step = _dst.step / sizeof(dst[0]);
 
+#if CV_SSE2
+    volatile bool useSIMD = checkHardwareSupport(CV_CPU_SSE);
```
Comment (ilya-lavrenov) [inline modules/imgproc/src/thresh.cpp]: should be `CV_CPU_SSE2` since you are using vectors of doubles.

Comment (ilya-lavrenov) [inline modules/imgproc/src/thresh.cpp]: @tomoaki0705, please, take this comment into account and we can merge your changes after.

Comment (alalek) [inline modules/imgproc/src/thresh.cpp]: BTW, why "volatile" is used here?

Comment (ilya-lavrenov) [inline modules/imgproc/src/thresh.cpp]: good question :) it's done so in other places of this file. I suppose it's just for consistency

**Comment by ilya-lavrenov** [comment]
:+1:

**Comment by tomoaki0705** [comment]
I appreciate your comment and your review.

**Pull request merged**